/**
 * MIT License
 *
 * Copyright (c) 2024 ylgeeker
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

syntax = "proto3";

package ylg.internal;

import "base.proto";

message Ping
{
    string _msg = 1;
}

message Pong
{
    int32  _code    = 1;
    string _errmsg  = 2;
    string _agentID = 3;
    string _msg     = 4;
}

message AuthRequest
{
    string _agentID = 1;
}

message AuthRespond
{
    int32  _code    = 1;
    string _errmsg  = 2;
    string _agentID = 3;
}

message RegisterAgentRequest
{
    string _ip         = 1;
    string _mac        = 2;
    string _netDevName = 3;
    string _machineID  = 4;
    string _hostName   = 5;
    string _agentID    = 6;
}

message RegisterAgentRespond
{
    int32    _code  = 1;
    string _errmsg  = 2;
    string _agentID = 3;
}

message UnregisterAgentRequest
{
    string _agentID = 1;
}

message UnregisterAgentRespond
{
    int32  _code    = 1;
    string _errmsg  = 2;
    string _agentID = 3;
}

message ReportAgentState
{
    string _agentID   = 1;
    AgentState _state = 2;
}

